scripts/base_classes.py

```diff
@@ -139,8 +140,24 @@ def get_bam_header(self, sample_id):
             ["samtools", "view", "-H", docker_bam], docker_image="stefpiatek/samtools:1.9", stdout=subprocess.PIPE
         )
         header = str(samtools.stdout, "utf-8")
+
+        for line in header.split("\n"):
+            for field_value in line.split():
+                if field_value.startswith("SM:"):
+                    sample_name = field_value.strip("SM:")
+        if not sample_name:
+            raise Exception(f"No SM tag found in file {docker_bam}")
+        assert (
+            sample_name == sample_id
+        ), f"sample-sheet sample_id '{sample_id}' for '{docker_bam}' does not match bam SM: '{sample_name}'"
+
         return header
 
+    def get_md5sum(self, file_path):
+        md5sum_proc = subprocess.run(["md5sum", file_path], check=True, stdout=subprocess.PIPE)
+        md5sum, path = str(md5sum_proc.stdout, "utf-8").split()
+        return md5sum, path
+

```

Comment:
Have you seen hashlib?

Reply:
Ooh I have not but that would've been a nicer way to do it